flashlight/fl/test/tensor/jit/JitEvaluatorTest.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <functional>
+
+#include <gtest/gtest.h>
+
+#include "flashlight/fl/tensor/Init.h"
+#include "flashlight/fl/tensor/Random.h"
+#include "flashlight/fl/tensor/Shape.h"
+#include "flashlight/fl/tensor/backend/af/ArrayFireBackend.h"
+#include "flashlight/fl/tensor/backend/af/ArrayFireTensor.h"
+#include "flashlight/fl/tensor/backend/jit/eval/Evaluator.h"
+#include "flashlight/fl/tensor/backend/jit/ir/ValueNode.h"
+
+using namespace fl;
+
+TEST(JitEvaluatorTest, evalValueNode) {
+  const auto tensor = fl::rand(Shape({2, 2}), dtype::f32);
+  const auto node = ValueNode::create(tensor.copy());
+  Evaluator evaluator(ArrayFireBackend::getInstance());
+  evaluator.eval(node);
+  ASSERT_TRUE(allClose(node->getResult().value(), tensor));
+  // node is owned locally (didn't transition to shared ownership)
+  delete node;
+}
+
+TEST(JitEvaluatorTest, evalScalarNode) {
+  Shape shape(Shape({2, 2}));
+  auto dtype = dtype::s32;
+  int val = 42;
+  const auto tensor = full(shape, val, dtype);
+  const auto node = ScalarNode::create(shape, dtype, val);
+  Evaluator evaluator(ArrayFireBackend::getInstance());
+  evaluator.eval(node);
+  ASSERT_TRUE(allClose(node->getResult().value(), tensor));
+  // node is owned locally (didn't transition to shared ownership)
+  delete node;
+}
+
+TEST(JitEvaluatorTest, evalBinaryNode) {
+  // c1  c2
+  //  \  /
+  //   add
+  Shape shape(Shape({2, 2}));
+  auto dtype = dtype::s32;
+  const auto c1 = ScalarNode::create(shape, dtype, 1);
+  const auto c2 = ScalarNode::create(shape, dtype, 2);
+  const auto add = BinaryNode::create(c1, c2, BinaryOp::Add);
```

Comment:
Note that I intentionally avoided using `JitTensor` to build up the IR here -- to keep the boundary between Evaluator and JitTensor (which again, serves as the parser).